pkg/workload/tpcc/tpcc.go

```diff
@@ -1014,13 +1028,21 @@ func (w *tpcc) Ops(
 	// Limit the amount of workers we initialize in parallel, to avoid running out
 	// of memory (#36897).
 	sem := make(chan struct{}, 100)
+	elemIndex := -1
 	for workerIdx := 0; workerIdx < w.workers; workerIdx++ {
 		workerIdx := workerIdx
 		var warehouse int
 		var p int
 		if len(w.multiRegionCfg.regions) == 0 {
-			warehouse = w.wPart.totalElems[workerIdx%len(w.wPart.totalElems)]
-			p = w.wPart.partElemsMap[warehouse]
+			for {
+				elemIndex = elemIndex + 1
+				warehouse = w.wPart.totalElems[elemIndex%len(w.wPart.totalElems)]
+				part := w.wPart.partElemsMap[warehouse]
+				if isMyPart(part) {
+					p = part
+					break
+				}
+			}
```

Comment:
1. Is this needed: `elemIndex%len(w.wPart.totalElems)`? Is this because `totalElems` is changed by a different go routine?

2. Can we make the for loop better with maybe something like
```
part := -1
for !isMyPart(part) {
        elemIndex = elemIndex + 1
				warehouse = w.wPart.totalElems[elemIndex%len(w.wPart.totalElems)]
				part := w.wPart.partElemsMap[warehouse]
}